bufferserver/src/main/java/com/datatorrent/bufferserver/internal/DataList.java

```diff
@@ -821,34 +848,36 @@ protected void release(boolean wait, boolean writer)
       final int refCount = this.refCount.decrementAndGet();
       if (canEvict(refCount, writer)) {
         assert (next != null);
-        final Runnable storer = getStorer(data, readingOffset, writingOffset, storage);
-        if (future != null && future.cancel(false)) {
-          logger.debug("Block {} future is cancelled", this);
-        }
-        final int numberOfInMemBlockPermits = DataList.this.numberOfInMemBlockPermits.get();
-        if (wait && numberOfInMemBlockPermits == 0) {
-          future = null;
-          storer.run();
-        } else if (numberOfInMemBlockPermits < MAX_COUNT_OF_INMEM_BLOCKS / 2) {
-          future = storageExecutor.submit(storer);
+        if (storage != null) {
```

Comment:
I refer to the case where the last subscriber is far behind Publisher for example after the downstream operator failure and re-subscription.
Consider moving the block inside `if (storage != null)` into a separate method like evictBlock(), so it will be easier to read the else condition.